### PR TITLE
Converts drowsy and eye blur to status effects, striking yet another two carbon level status vars

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

You know the deal by now.

- Drowsiness is now tracked via status effect.
- Eye blue is now tracked via status effect.

In converting these over, cleaned up a bit of code relating to some other effects. Attempts to unify behavior between some of them, namely certain biotypes or mob types shouldn't be experiencing certain effects. 

## Why It's Good For The Game

More stuff moved to status effects, slightly more cleaner and better to work with code. Allows for all mobs that can sleep to be able to get drowsy, too. 

## Changelog

:cl: Melbert
refactor: Drowsiness and Blurred Eyes are now tracked via status effect.
/:cl:

### CODE REVIEW

File: code/datums/status_effects/debuffs/screen_blur.dm [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	tick_interval = 1 SECONDS
+	alert_type = null
+	remove_on_fullheal = TRUE
+
+/datum/status_effect/eye_blur/on_creation(mob/living/new_owner, duration = 10 SECONDS)
+	src.duration = duration
+	return ..()
+
+/datum/status_effect/eye_blur/on_apply()
+	if(owner.mob_biotypes & (MOB_ROBOTIC|MOB_SPIRIT|MOB_EPIC))
```
Comment (JohnFulpWillard) [inline code/datums/status_effects/debuffs/screen_blur.dm:17]: Use ``NO_HALLUCINATION_BIOTYPES`` here?

File: code/datums/status_effects/debuffs/screen_blur.dm [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	update_blur()
+	return TRUE
+
+/datum/status_effect/eye_blur/on_remove()
+	UnregisterSignal(owner, COMSIG_MOB_LOGIN)
+	if(!owner.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = owner.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+	game_plane_master_controller.remove_filter("eye_blur")
```
Comment (ChungusGamer666) [inline code/datums/status_effects/debuffs/screen_blur.dm:32]: Move this to using the game plate, why not?

**Comment by Jacquerel** [comment]
Will this fix the issue where aheal sometimes leaves people with monochrome vision or is that unrelated?

**Comment by Ghommie** [comment]
> Will this fix the issue where aheal sometimes leaves people with monochrome vision or is that unrelated?

That's related to the screen colour datums, which I believe need to go now that we have game plates and filters.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 3]
```diff
@@ -658,7 +658,7 @@
 
 /datum/reagent/medicine/oculine/on_mob_life(mob/living/carbon/affected_mob, delta_time, times_fired)
 	affected_mob.adjust_blindness(-2 * REM * delta_time * normalise_creation_purity())
-	affected_mob.adjust_blurriness(-2 * REM * delta_time * normalise_creation_purity())
+	affected_mob.adjust_eye_blur(-4 SECONDS * REM * delta_time * normalise_creation_purity())
```
Comment (Kylerace) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:661]: im not sure if this is correct. the original version didnt have a timer by default afaict and this one does, but it keeps the process loop used in the old one. wouldnt this increase rate at which blurriness lowers over time?

File: code/datums/status_effects/debuffs/screen_blur.dm [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+/// Updates the blur of the owner of the status effect.
+/// Also a signal proc for [COMSIG_MOB_LOGIN], to trigger then when the mob gets a client.
+/datum/status_effect/eye_blur/proc/update_blur(datum/source)
+	SIGNAL_HANDLER
+
+	if(!owner.hud_used)
+		return
+
+	var/time_left_in_seconds = (duration - world.time) / 10
```
Comment (Fikou) [inline code/datums/status_effects/debuffs/screen_blur.dm:46]: can we make this / 1 SECONDS instead?

**Review by Fikou — APPROVED** [review]

**Pull request merged**